weblate/machinery/deepl.py

```diff
@@ -24,6 +24,13 @@ class DeepLTranslation(MachineTranslation):
     hightlight_syntax = True
     settings_form = DeepLMachineryForm
 
+    def __init__(self, settings: dict[str, str]):
+        super().__init__(settings)
+        self.formality = (
+            self.settings["formality"] if "formality" in self.settings else "default"
+        )
+        self.mtid = self.mtid + "_" + self.formality
+
```

Comment:
I'd move this out of init:

```suggestion
```

* formality can be handled at request time
* `mtid` is used for caching list of languages and rate limiting, none of these is affected by formality

Reply:
Should I just commit this suggestion with the others or did you just select/highlight this code in order to comment? I don't see where the `self.mtid` field would be written in your suggested change

Reply:
So should I overwrite that method in `deepl.py` as follows?
```
def translate_cache_key(self, source, language, text, threshold):
    if not self.cache_translations:
        return None
    return super().translate_cache_key(source, language, text, threshold) + ":" + self.settings.get("formality", "default")
```

Reply:
I have pushed another commit with that change.
I also moved the check for the `cache_translations` variable to the `get_cached` method to avoid code duplication